File: main/src/main/java/net/fwitz/math/main/binary/dual/functions/trig/DualSinZ.java
package net.fwitz.math.main.binary.dual.functions.trig;

import net.fwitz.math.binary.dual.DualNumber;
import net.fwitz.math.plot.binary.dual.DualNumberFunctionPlot;

public class DualSinZ {
    public static void main(String[] args) {
        new DualNumberFunctionPlot("sin(z)")
                .computeFn(DualNumber::sin)
                .render();
    }
}

File: main/src/main/kotlin/net/fwitz/math/main/binary/dual/functions/trig/DualSinZ.kt
package net.fwitz.math.main.binary.dual.functions.trig

import net.fwitz.math.plot.binary.dual.DualNumberFunctionPlot

object DualSinZ {
    @JvmStatic
    fun main(args: Array<String>) = DualNumberFunctionPlot("sin(z)")
        .computeFn { z -> z.sin }
        .render()
}